packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts

```diff
@@ -268,6 +268,80 @@ describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) =
             assert.strictEqual(response?.status, 200, `Expecting a 200 response`);
             assert(container2.closed !== true, "Container should not have closed");
         });
+
+        /**
+         * Function that inverses the does not reject condition, i.e., asserts for reject instead. Used in these tests
+         * to demonstrate that because of a bug we are not getting the expected results. Once the bug is fixed, these
+         * asserts should start working as expected.
+         */
+        async function assertDoesNotRejectInverse(block: Promise<any>, message?: string | Error) {
+            return assert.rejects(block, message);
+        };
+
+        /**
+         * This test validates that when blobs are de-duped in different containers, these containers can use these
+         * blobs irrespective of whether the original blob is tombstoned. Basically, after uploading a blob, a container
+         * should be able to use it the same way whether it was de-duped or not.
+         * Note: Currently this fails because the behavior is not as expected and so it's expecting an error.
+         */
+        itExpects("should allow access to blobs that are de-duped in different containers",
+        [
+            { eventName: "fluid:telemetry:BlobManager:GC_Tombstone_Blob_Requested" }
+        ],
+        async () => {
+            const { dataStore: mainDataStore, summarizer } = await createDataStoreAndSummarizer();
+
+            // Upload an attachment blob.
+            const blobContents = "Blob contents";
+            const blobHandle = await mainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Reference and then unreference the blob so that it's unreferenced in the next summary.
+            mainDataStore._root.set("blob1", blobHandle);
+            mainDataStore._root.delete("blob1");
+
+            // Summarize so that the above attachment blobs are marked unreferenced.
+            await provider.ensureSynchronized();
+            const summary1 = await summarizeNow(summarizer);
+
+            // Wait for half sweep timeout and load a container. This container will upload a blob with the same content
+            // as above so that it is de-duped. This container should be able to use this blob until its session
+            // expires.
+            await delay(sweepTimeoutMs / 2);
+            const container2 = await loadContainer(summary1.summaryVersion);
+            const container2MainDataStore = await requestFluidObject<ITestDataObject>(container2, "default");
+            // Upload the blob and keep the handle around until the blob uploaded by first container is tombstoned.
+            const container2BlobHandle =
+                await container2MainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Wait for sweep timeout so that the blob uploaded by the first container is tombstoned.
+            await delay(sweepTimeoutMs + 10);
```

Comment:
nit: `sweepTimeoutMs / 2 + 10` right?  What is SessionExpiry set to and will this trigger that? I guess not.

Reply:
yes, you are right. missed that part!